mypy/checker.py

```diff
@@ -3568,6 +3568,20 @@ def lookup_qualified(self, name: str) -> SymbolTableNode:
                 msg = "Failed qualified lookup: '{}' (fullname = '{}')."
                 raise KeyError(msg.format(last, name))
 
+    def lookup_fully_qualified_or_none(self, fullname: str) -> Optional[SymbolTableNode]:
```

Comment:
I'd rather not add yet another lookup function, even if we are planning to refactor this soon. Can you move this to a new module, such as `mypy.lookup`? This can be a top-level function that accepts a modules dictionary and fullname. You can then pass the modules dictionary to the plugin, and the plugin can call this function (also the semantic analyzer can call the new function).

Passing the modules dictionary to plugins would have a few additional benefits: it only needs to be done once per build, and the plugins would be given less access to semantic analyzer and type checker internals.

Reply:
Yes, this makes sense. I incidentally pushed before fixing some things, I will one more commit soon.